File: android/app/src/main/java/com/dailymemory/domain/usecase/memory/UpdateMemoryUseCase.kt
package com.dailymemory.domain.usecase.memory

import com.dailymemory.domain.model.Memory
import com.dailymemory.domain.repository.MemoryRepository
import javax.inject.Inject

/**
 * Use case for updating an existing memory.
 */
class UpdateMemoryUseCase @Inject constructor(
    private val memoryRepository: MemoryRepository
) {
    suspend operator fun invoke(memory: Memory): Result<Memory> {
        return try {
            memoryRepository.update(memory)
            Result.success(memory)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
}

File: ios/DailyMemory/Domain/UseCase/Memory/UpdateMemoryUseCase.swift
import Foundation

/// Use case for updating an existing memory.
final class UpdateMemoryUseCase {
    private let memoryRepository: MemoryRepository

    init(memoryRepository: MemoryRepository = MemoryRepositoryImpl()) {
        self.memoryRepository = memoryRepository
    }

    func execute(_ memory: Memory) async throws -> Memory {
        try await memoryRepository.update(memory)
        await SyncManager.shared.enqueueMemoryChange(id: memory.id, type: .update)
        return memory
    }
}
